specs/IsSwipeNavigationEnabled.md

```diff
@@ -0,0 +1,96 @@
+# Background
+
+Swiping navigation on touch screen includes: 
+1. Swipe left/right (swipe horizontally) to navigate to previous/next page in navigation history. 
+1. Pull to refresh (swipe vertically) the current page.
+
+In response to customer requests, the WebView2 team has introduced this Swipe Navigation API which allows developers to disable or enable the the ability of end users to use swiping gesture to navigate in WebView via a setting.
+
+In this document we describe the new setting. We'd appreciate your feedback.
+
+# Description
+The default value for `IsSwipeNavigationEnabled` is `TRUE`.
+When this setting is set to `FALSE`, it disables the ability of the end users to use swiping gesture on touch input enabled devices to trigger navigations such as swipe left/right to go back/forward or pull to refresh current page.
+Disabling/Enabling `IsSwipeNavigationEnabled` does not take effect until the next navigation.
+
+
+# Examples
+```cpp
+wil::com_ptr<ICoreWebView2> webView;
+void SettingsComponent::ToggleSwipeNavigationEnabled()
+{
+    // Get webView's current settings
+    wil::com_ptr<ICoreWebView2Settings> coreWebView2Settings;
+    CHECK_FAILURE(webView->get_Settings(&coreWebView2Settings));
+
+    wil::com_ptr<ICoreWebView2Settings6> coreWebView2Settings6;
+    coreWebView2Settings6 = coreWebView2Settings.try_query<ICoreWebView2Settings6>();
+    if(coreWebView2Settings6) 
+    {
+        BOOL enabled;
+        CHECK_FAILURE(coreWebView2Settings6->get_IsSwipeNavigationEnabled(&enabled));
+        CHECK_FAILURE(coreWebView2Settings6->put_IsSwipeNavigationEnabled(enabled ? FALSE : TRUE));
+    }
+}
+```
+
+```c#
+private WebView2 _webView;
+void ToggleSwipeNavigationEnabled()
+{
+    var coreWebView2Settings = _webView.CoreWebView2.Settings;
+    coreWebView2Settings.IsSwipeNavigationEnabled = !coreWebView2Settings.IsSwipeNavigationEnabled;
+}
+```
+
+# Remarks
+This API only affects the decision whether to trigger navigations of the main html page when end users perform overscrolling motions by swiping and has no effect on the scrolling interaction used to explore the web content shown in WebView2 or the CSS overscroll-behavior property.
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```cpp
+[uuid(45b1f964-f703-47ac-a19a-b589dd0c5559), object, pointer_default(unique)]
+interface ICoreWebView2Settings6 : ICoreWebView2Settings5 {
+  /// Swiping gesture navigation on touch screen includes:
+  /// 1. Swipe left/right (swipe horizontally) to navigate to previous/next
+  /// page in navigation history.
+  /// 2. Pull to refresh (swipe vertically) the current page. (This feature is
+  /// currently disabled by default in the browser, to enable in WebView2, use
+  /// `put_AdditionalBrowserArguments` api with "--pull-to-refresh" switch).
```

Comment:
Checking: this won't be necessary anymore with this new API?

Reply:
I think this is saying "Even if you enable swipe navigation, pull-to-refresh won't work unless you also enable it via `put_AdditionalBrowserArguments()`." This property is really a reverse-sense "IsSwipeNavigationDisabled" in that there are many things that disable it, and you have to get them all to allow pull-to-refresh for it to work.

I think that even with this behavior, the current name is okay - we need to call out in the docs

> If this property is `TRUE`, then all configured navigation gestures are enabled.
>
> * Swiping left and right to navigate forward and backward is always configured.
> * Swiping down to refresh requires the `--pull-to-refresh` option to be included in the additional browser arguments. (See `put_AdditionalBrowserArguments`.)

Is "refresh" considered a "navigation" by the WebView2? (If I refresh the page, are navigation events raised? Do properties marked as "takes effect at the next navigation" take effect?)

Reply:
> Is "refresh" considered a "navigation" by the WebView2? (If I refresh the page, are navigation events raised? Do properties marked as "takes effect at the next navigation" take effect?)
Yes: yes to navigation events and yes to takes effect at the next navigation.

Please update the docs per Raymond's notes